### PR TITLE
Fix property getter with custom return type

### LINKED ISSUE #68001: [4.0] Can't use existing class function with a custom return type for property getter
### Godot version

Godot 4.0 beta 3

### System information

Windows 10

### Issue description

GDScript marks an error when using existing class function with a custom Object return type for property getter, this is despite both the property and the function return type being the same custom type. The error marked is along the lines of:
`Function with return type "GDScript" cannot be used as getter for a property of type "{NameOfCustomTypeHere}".`

As seen below:
image

### Steps to reproduce

1. Open example project
2. Open example.gd
3. Errors at each variable declaration using existing class function with a custom return type for property getter:
```
Line 3: Function with return type "GDScript" cannot be used as getter for a property of type "CustomObject".
Line 8: Function with return type "GDScript" cannot be used as getter for a property of type "CustomNode".
```

### Minimal reproduction project

[example.zip](https://github.com/godotengine/godot/files/9892910/example.zip)

### PR DESCRIPTION
**Pull request by adamscott**
# TL;DR
Makes it possible to use custom return types with property getters.

# Minimal reproductible project
[example.zip](https://github.com/godotengine/godot/files/9892910/example.zip) (straight out of #68001)

# Fixes
Fixes #68001 - [4.0] Can't use existing class function with a custom return type for property getter

### CODE REVIEW

File: modules/gdscript/gdscript_analyzer.cpp [hunk 1]
```diff
@@ -985,21 +985,26 @@ void GDScriptAnalyzer::resolve_class_body(GDScriptParser::ClassNode *p_class) {
...
-
-					} else if (getter_function->parameters.size() != 0 || getter_function->datatype.has_no_type()) {
-						push_error(vformat(R"(Function "%s" cannot be used as getter because of its signature.)", getter_function->identifier->name), member.variable);
-
-					} else if (!is_type_compatible(member.variable->datatype, getter_function->datatype, true)) {
-						push_error(vformat(R"(Function with return type "%s" cannot be used as getter for a property of type "%s".)", getter_function->datatype.to_string(), member.variable->datatype.to_string()), member.variable);
-
 					} else {
-						has_valid_getter = true;
+						GDScriptParser::DataType return_datatype = getter_function->datatype;
```
Comment (adamscott) [inline modules/gdscript/gdscript_analyzer.cpp:989]: This is kinda hard to see, but I pushed all the else if inside the big else of `getter_function != nullptr`. Then, I assign the `return_datatype` variable with the actual `return_type`, if it's not null. It is not considered at all in the master branch.

**Review by akien-mga — APPROVED** [review]
Looks good to me.

**Comment by akien-mga** [comment]
Thanks!
